enclave/core/CMakeLists.txt

```diff
@@ -95,6 +95,47 @@ if (OE_SGX)
   set_source_files_properties(sgx/td_basic.c PROPERTIES COMPILE_FLAGS
                                                         -fno-stack-protector)
 
+  # Switch compiler from MSVC to Clang for compiling enclave libraries
+  if (WIN32 AND MSVC)
+    find_program(CLANG_C_COMPILER
+                 NAMES clang
+                 PATHS "C:/Program Files/LLVM/bin"
+                 NO_DEFAULT_PATH
+                 REQUIRED)
+    execute_process(
+      COMMAND ${CLANG_C_COMPILER} --version
+      OUTPUT_VARIABLE CLANG_C_COMPILER_VERSION
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if (CLANG_C_COMPILER_VERSION MATCHES "clang version ([0-9]+\.[0-9]+\.[0-9]+)")
+      set(CLANG_C_COMPILER_VERSION ${CMAKE_MATCH_1})
+    else ()
+      message(
+        FATAL_ERROR
+        "Could not determine Clang version from ${CLANG_C_COMPILER}"
+      )
+    endif ()
+    set(CMAKE_C_COMPILER ${CLANG_C_COMPILER})
+    set(CMAKE_C_COMPILER_VERSION ${CLANG_C_COMPILER_VERSION})
```

Comment:
Should we enforce clang-11 check here?

Reply:
Good idea, it has not been enforced anywhere in this CMakeList. Will make the change. Thanks!